tests/tset_password.py

```diff
@@ -0,0 +1,73 @@
+#tests/test_password.py
+import unittest
+import json
+from api import create_app
+
+
+class UserLogoutTestCase(unittest.TestCase):
+    """User logout taste cases class ."""
+
+    def setUp(self):
+        """Set up test variables."""
+        self.app = create_app(config_name="testing")
+        # initialize the test client
+        self.client = self.app.test_client
+        # This is the user test json data with a predefined username, email and password
+        
+        self.user_data = {
+            'user_email': 'looky@example.com',
+            'password': 'testexample'
+        }
+        self.user_data_2 = {
+            'user_email': 'example@example.com',
+            'password': 'test_123'
+        }
+
+    def test_user_can_reset_password(self):
+        """Test user can change there password given correct credentials."""
+        response = self.client().post('/api/v1/auth/signup', data=self.user_data)
+        self.assertEqual(response.status_code, 201)
+        login_response = self.client().post('/api/v1/auth/login', data=self.user_data)
+        self.assertEqual(login_response.status_code, 200)
+        #Define header dictionary
+        access_token = json.loads(login_response.data.decode())['access_token']
+        reset_password = {
+            "old_password": "testexample",
+            "new_password": "123456"
+        }
+        reset_response = self.client().post('/api/v1/auth/reset-password',
+                                       headers=dict(Authorization='Bearer ' + access_token), data=reset_password)
+        #assert that the status code is equal to 200
+        self.assertEqual(reset_response.status_code, 200)
+        #return result in json format
+        result = json.loads(reset_response.data.decode())
+        #test that the response contains success message
+        self.assertEqual(result["message"],
+                         "You have successfully reset your password.")
+        
+
+    def test_user_cannot_reset_password_with_invalid_credential(self):
+        """Test user cannot change there password given incorrect credentials."""
+        response = self.client().post('/api/v1/auth/signup', data=self.user_data_2)
+        self.assertEqual(response.status_code, 201)
+        login_response = self.client().post('/api/v1/auth/login', data=self.user_data_2)
+        self.assertEqual(login_response.status_code, 200)
```

Comment:
I suggest you don't test endpoints in another endpoint. E.g:
- The `test_user_cannot_reset_password_with_invalid_credential` should not be testing other endpoints too. Having `assert` for the signup and login requests I think is not good

Reply:
I thought it was wise to show how the whole process goes in each case.

Reply:
Consider it's a test case. In order to know what is failing, it's better not to mix them up in one test case

Reply:
For example your `reset_password` feature is working properly but your `login` is not.
Are you aware you'll get `reset_password` test errors.
This may mislead you into trying to debug the working `reset_password`

Reply:
Got it, let me implement that.